packages/run-protocol/src/vpool-xyk-amm/pool.js

```diff
@@ -252,6 +250,7 @@ const poolBehavior = {
       ),
     );
     state.zcf.reallocate(userSeat, poolSeat);
+    state.liqTokenSupply -= liquidityValueIn;
```

Comment:
No, the issue is that it's an accounting number that isn't part of the reallocate. If it were included in the reallocate, then it could have been atomic with the other state changes.

Since it's tracking the outstanding liquidity (held by liquidity providers) it doesn't seem possible to have zoe manage conservation and atomicity.